cfme/services/catalogs/catalog.py

```diff
@@ -1,115 +1,110 @@
+        return (
+            self.title.text == "Adding a new Catalog" and
+            self.catalogs.is_opened
+        )
 
-item_multiselect = web_ui.MultiSelect(
-    "//select[@id='available_fields']",
-    "//select[@id='selected_fields']",
-    "//div[@id='column_lists']//a[contains(@href, 'button=right')]/img",
-    "//div[@id='column_lists']//a[contains(@href, 'button=left')]/img")
 
-form = web_ui.Form(
-    fields=[('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('button_multiselect', item_multiselect),
-            ('add_button', form_buttons.add),
-            ('save_button', form_buttons.save),
-            ('cancel_button', form_buttons.cancel)])
+class EditCatalogView(CatalogForm):
 
-item_form = web_ui.Form(
-    fields=[('type_select', "//select[@id='st_prov_type']"),
-            ('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('display_checkbox', Input("display")),
-            ('add_button', form_buttons.add)])
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Editing Catalog{}".format(self.obj.name) and
+            self.catalogs.is_opened
+        )
 
-match_page = partial(match_location, controller='catalog', title='Catalogs')
 
+class DetailsCatalogView(ServicesCatalogView):
 
-def _all_catalogs_add_new(_):
-    if BZ(1213863).blocks:
-        sel.pytest.skip("Blocker on the bug 1213863")
-    catalog_tree('All Catalogs')
-    cfg_btn('Add a New Catalog')
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == 'Catalog "{}"'.format(self.obj.name) and
+            self.catalogs.is_opened
+        )
 
 
 class Catalog(Updateable, Pretty, Navigatable):
-    """Represents a Catalog"""
-    pretty_attrs = ['name', 'items']
 
     def __init__(self, name=None, description=None, items=None, appliance=None):
+        Navigatable.__init__(self, appliance)
         self.name = name
         self.description = description
         self.items = items
-        Navigatable.__init__(self, appliance=appliance)
-
-    def __str__(self):
-        return self.name
 
     def create(self):
-        navigate_to(self, 'Add')
-        # For element not found exception (To be removed)
-        sel.sleep(5)
-        web_ui.fill(form, {'name_text': self.name,
-                           'description_text': self.description,
-                           'button_multiselect': self.items},
-                    action=form.add_button)
-        flash_str = 'Catalog "{}" was saved'
-        flash.assert_success_message(flash_str.format(self.name))
+        view = navigate_to(self, 'Add')
+        view.fill({'name': self.name,
+                   'description': self.description})
+        view.add_button.click()
+        view.flash.assert_success_message('Catalog "{}" was saved'.format(self.name))
```

Comment:
So, it should actually create a view that it is supposed to have landed on and then assert is_displayed on it. Then use that view for checking flash. Something like:

```python
view.fill(...)
view.add_button.click()
view = self.create_view(ViewItShouldHaveLandedOn)  # In case of update(self, updates) you should also pass override=updates
assert view.is_displayed
view.flash.assert_no_error()
view.flash.assert_success_message('Catalog "{}" was saved'.format(self.name))
```

create_view passes self automatically in the view as the context object.

From the latest widgetastic version you can use fill_with which triggers the appropriate actions upon filling:
```python
if view.fill_with({...}, on_change=view.add_button, no_change=view.cancel_button):
    view = self.create_view(ViewItShouldHaveLandedOn)  # In case of update(self, updates) you should also pass override=updates
    assert view.is_displayed
    view.flash.assert_no_error()
    view.flash.assert_success_message('Catalog "{}" was saved'.format(self.name))
else:  # when cancel
    view = self.create_view(AnotherViewItShouldHaveLandedOn)  # In case of update(self, updates) DO NOT pass updates here
    assert view.is_displayed
    view.flash.assert_no_error()
    view.flash.assert_success_message('Add was cancelled')
```

@psav is this the way how we should go forward? More checking? I do it like this with automate.

Also the self.name, in some places the display names, descriptions and such things change the used string based on what is and what is not specified. I think that since this is a conversion it should not be the case but just reminding :)